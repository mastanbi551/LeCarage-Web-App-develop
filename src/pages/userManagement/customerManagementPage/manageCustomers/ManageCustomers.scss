@import "../../../../styles//variables.scss";

.details-card-containers {
  input {
    width: 100%;
    color: $primaryBlack !important;
    box-sizing: border-box;
    width: 380px;
    height: 40px;

    border: 1px solid $lightgrey;
  }

  .label {
    @include label;
  }

  &__user-id-name-details,
  &__user-email-cellnumber-details,
  &__payment-details {
    display: flex;
    flex-direction: row;
    width: 100%;

    &__user-type-id-container,
    &__user-type-name-container,
    &__car-holder-container,
    &__bank-name-container {
      flex-direction: column;
      flex: 50%;
      text-align: left;
      padding-right: 40px;
      text-align: left;

      .label {
        @include label;
      }

      input {
        width: 100% !important;
        color: $primaryBlack !important;
        box-sizing: border-box;
        width: 380px;
        height: 40px;

        border: 1px solid $lightgrey;
      }
    }
  }

  &__user-address-container {
    padding: 2% 0;
    text-align: left;

    &__heading {
      @include SubHeading2;
    }
  }

  &__user-address-container {
    &__address-details {
      &__delete-icon {
        text-align: right;
        padding-right: 40px;
      }

      &__add-Address {
        display: flex;
        flex-direction: row;
        width: 100%;

        input {
          width: 100%;
        }
      }
    }
  }

  .no-wrap {
    flex-wrap: nowrap !important;
    display: flex;
  }

  &__cars-container {
    &__cars-details {
      &__add-car {
        display: flex;
        flex-direction: row;
        width: 100%;
      }

      &__car-heading-delete-icon {
        display: block;
        position: relative;

        .label {
          display: inline-block;
          float: left;
          @include label;
        }

        svg {
          display: inline-block;
          float: right;
          position: absolute;
          right: 10px;
        }
      }

      &__table-info {
        thead {
          border: 1px solid $lightgrey;
        }

        tbody {
          background: $secondaryWhite;
        }

        .input-text-field .input-field {
          color: $primaryBlack;
        }
      }
    }
  }
}

.manage-customers-page {
  text-align: left;

  input {
    width: 100%;
    padding: 10px;
  }

  .label {
    @include label;
  }
}

.overflow {
  overflow: auto;
  height: 430px;
}

.modal-view {
  //padding-top: 50%;
}

.add-car-btn {
  margin-right: 60%;
  box-sizing: border-box;
  width: 16%;
  height: 38px;
  color: $primaryGreen;
  background-color: $White;
  border: 2px solid $primaryGreen;
}

.modal-heading-row {
  left: 20.49%;
  right: 20.49%;
  top: 46.67%;
  bottom: 51.22%;
  background: $White;
  box-shadow: inset 0px 1px 0px $secondaryLightGrey;
}

.modal-list {
  left: 20.49%;
  right: 20.49%;
  top: 46.67%;
  bottom: 49.49%;
  background: $primaryWhite;
  box-shadow: inset 0px 1px 0px $secondaryLightGrey;
}

.modal-save-btn {
  left: 0%;
  right: 0%;
  top: 0%;
  bottom: 0.79%;
  background: $primaryGreen;
  width: 8%;
  height: 38px;
  border: 0px none;
}

.modal-cancel-btn {
  box-sizing: border-box;
  left: 0%;
  right: 0%;
  top: 0%;
  width: 10%;
  height: 38px;
  bottom: 0.79%;
  border: 2px solid $primaryGreen;
  background-color: $White;
  color: $primaryGreen;
}

.overflwtable {
  overflow-y: auto;
  height: 300px;
}

.checkbox-btn {
  color: $primaryBlack;
}

input[type="checkbox"]:checked::before {
  content: "âœ“";
  font-size: 14px;
  font-weight: 800;
  color: $primaryWhite;
  position: relative;
  top: -3px;
  left: 1px;
  width: 15px;
  height: 15px;
}

input[type="checkbox"] {
  border: 1px solid $primaryBlack;
}
.checkbox-text {
  padding-left: 2px;
}

.dropdown {
  margin-bottom: 1px !important;
  padding: 0px;
}

.customerspassword .input-text-field span.multiple-icons i:last-child svg {
  color: $primaryBlack;
}

.mgrght{
  margin-right: 27px;
}


.car_Search{
  margin-left: 10px;
    border-radius: 0px;
    border: 1px solid #cccc;
    padding: 5px;
}
::placeholder{
  font-size: 14px;
}

