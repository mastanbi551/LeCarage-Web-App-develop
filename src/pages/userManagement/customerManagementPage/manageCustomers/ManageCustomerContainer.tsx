import React, { useEffect, useState } from 'react';
import { useLocation, useNavigate } from 'react-router-dom';
import { ManageCustomers } from './ManageCustomers';
import { useAppDispatch, useAppSelector } from '../../../../hooks/Hooks';

import _ from 'lodash';
import {
  reset,
  updateCustomerRequest,
  viewUnLinkedCarCustomerRequest,
} from '../customerManagementPageState/CustomerManagementSlice';
import { toast } from 'react-toastify';
import { getStatusDetails } from '../../../commonStates/CommonSlice';
import {
  createCustomerRequest,
  viewCustomerRequest,
} from '../customerManagementPageState/CustomerManagementSlice';
import { NotFound } from '../../../notFound/NotFound';
import { AccessTypes, responseStatus, routeNames } from '../../../../utils/constants/ConstantStrings';
import { uploadImage } from '../../../../utils';
import { Loader } from '../../../../components';

export interface CarIProps {
  carId: string;
  Color: string;
  Make: string;
  Model: string;
  'Number Plate': string;
  checked: boolean;
  additionalNotes: string;
}

export interface CustomerInfoProps {
  uId: string;
  profile: {
    email: string;
    phoneNumber: string;
    firstName: string;
    lastName: string;
    statusId: string;
    password: string;
  };
  address: {
    addressLineOne: string;
    streetName: string;
    houseNumber: string;
    postalCode: string;
    cityName: string;
  }[];
  cars: {
    carId: string;
  }[];
}

export const ManageCustomerContainer = () => {
  const dispatch = useAppDispatch();
  const navigate = useNavigate();
  const location = useLocation();
  const [modalShow, setModalShow] = useState(false);

  //to check the access
  const { routes } = useAppSelector((state) => state.sidebarReducer);
  const [access, setAccess] = useState('');
  useEffect(() => {
    if (access === '' && routes) {
      const SubRoutes = routes ? _.get(routes, routeNames.userManagement) : 0;
      if (SubRoutes) {
        const getAccess = _.filter(
          SubRoutes,
          (obj: {
            accessId: string;
            featureAccess: string;
            isParent: boolean;
            label: string;
            parentFeature: string;
            value: string;
          }) => obj.label === routeNames.customers
        );
        if (getAccess.length > 0) {
          setAccess(getAccess ? getAccess[0].featureAccess : '');
        }
      }
    }
  }, [access, routes]);
  //to check the access

  const { statusList } = useAppSelector((state) => state.commonReducer);

  const [customerInfo, setCustomerInfo] = useState(
    location.state ? location.state.customerInfo : ''
  );

  const [isCreate, setIsCreate] = useState(
    location.state && location.state.action === 'create' ? true : false
  );

  const [isDisable, setIsDisable] = useState(
    location.state &&
      location.state.customerInfo &&
      location.state.action === 'view'
      ? true
      : false
  );

  const [isEditable, setIsEditable] = useState(
    location.state &&
      location.state.customerInfo &&
      location.state.action === 'edit'
      ? true
      : false
  );

  const [statusDropdownData, setStatusDropdownData] = useState<
    { label: string; value: string }[]
  >([]);

  const {
    unlinkedCarIsLoading,
    linkedCarIsLoading,
    viewLinkedCustomerResponseData,
    viewUnLinkedCustomerResponseData,
    viewCustomerIsLoading,
    viewCustomerResponseData,
    isLoading,
    responseData,
    deleteCustomerIsLoading,
    deleteCustomerResponseData,
  } = useAppSelector((state) => state.customerManagementReducer);

  const [customerData, setCustomerData] = useState({
    uId: '',
    profile: {
      email: '',
      phoneNumber: '',
      password: '',
      firstName: '',
      lastName: '',
      statusId: '',
      photoURL: '',
    },
    address: [
      {
        addressLineOne: '',
        streetName: '',
        houseNumber: '',
        postalCode: '',
        cityName: '',
      },
    ],
    cars: [
      {
        carId: '',
        additionalNotes: '',
        Make: '',
        Color: '',
        Model: '',
        'Number Plate': '',
      },
    ],
  });
  const [passwordType, setPasswordType] = useState('password');
  const [unlinkedCars, setUnLinkedCars] = useState<CarIProps[]>([]);
  const [linkedCars, setLinkedCars] = useState<CarIProps[]>([]);

  const [photoURL, setPhotoURL] = useState('');

  const onDrop = async (file: File) => {
    const webUrl = await uploadImage(file);
    if (webUrl !== false) {
      setPhotoURL(webUrl[0]);
    }
  };

  const [tempCheckedCars, setTempCheckedCars] = useState<CarIProps[]>([]);

  const [tempEditCheckedCars, setTempEditCheckedCars] = useState<CarIProps[]>(
    []
  );

  const [checkedCars, setCheckedCars] = useState<CarIProps[]>([]);
  const [editCheckedCars, setEditCheckedCars] = useState<CarIProps[]>([]);

  const [userDetailsCollapse, setUserDetailsCollapse] = useState(true);
  const [carsDetailsCollapse, setCarsDetailsCollapse] = useState(true);

  const [searchValue, setSearchValue] = useState('');

  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
    setSearchValue(event.target.value);
  };

  useEffect(() => {
    if (
      customerData &&
      customerData.uId === '' &&
      customerInfo !== '' &&
      (isEditable || isDisable)
    ) {
      dispatch(reset());
      const params = { uId: _.get(customerInfo, 'uId') };
      dispatch(viewCustomerRequest(params));
      setCustomerInfo('');
    }
  }, [customerData, customerInfo]);

  useEffect(() => {
    if (isLoading) {
      if (responseData.status === responseStatus.success) {
        dispatch(reset());
        navigate('/customers');
      } else if (responseData.status === responseStatus.fail) {
        toast.error(responseData.errDescription, {
          position: toast.POSITION.TOP_CENTER,
          className: 'toast-message',
        });
      }
    }
  }, [isLoading, responseData]);

  useEffect(() => {
    if (unlinkedCars.length == 0) {
      dispatch(viewUnLinkedCarCustomerRequest());
    }
  }, [unlinkedCars]);

  useEffect(() => {
    if (viewCustomerIsLoading) {
      if (viewCustomerResponseData.status === responseStatus.success) {
        setCustomerData(viewCustomerResponseData.details[0]);
        const setChecked = _.map(
          viewCustomerResponseData.details[0].cars,
          (obj) => {
            return { ...obj, checked: true };
          }
        );

        setPhotoURL(viewCustomerResponseData.details[0].profile.photoURL);
        setTempEditCheckedCars(setChecked);
        setEditCheckedCars(setChecked);
      }
    }
    if (deleteCustomerIsLoading) {
      if (deleteCustomerResponseData.status === responseStatus.success) {
        navigate('/customers');
      }
    }
  }, [
    viewCustomerIsLoading,
    viewCustomerResponseData,
    deleteCustomerIsLoading,
    deleteCustomerResponseData,
  ]);

  useEffect(() => {
    if (unlinkedCarIsLoading) {
      if (viewUnLinkedCustomerResponseData.status === responseStatus.success) {
        const updatedCars = _.map(viewUnLinkedCustomerResponseData.details, car => ({...car, checked: false}));
        setUnLinkedCars(updatedCars);
      }
    }
    if (linkedCarIsLoading) {
      if (viewLinkedCustomerResponseData.status === responseStatus.success) {
        setLinkedCars(viewLinkedCustomerResponseData.details);
      }
    }
  }, [
    unlinkedCarIsLoading,
    viewUnLinkedCustomerResponseData,
    linkedCarIsLoading,
    viewLinkedCustomerResponseData,
  ]);

  const handleModalClose = () => setModalShow(false);
  const handleModalOpen = () => {
    setTempCheckedCars([]);
    setModalShow(true);
    setSearchValue('');
  };

  const handlePasswordTypeChange = () => {
    if (passwordType === '') setPasswordType('password');
    else setPasswordType('');
  };

  const handleUserDetailsCollapse = () => {
    setUserDetailsCollapse(!userDetailsCollapse);
  };

  const handleCarsDetailsCollapse = () => {
    setCarsDetailsCollapse(!carsDetailsCollapse);
  };

  const backButtonHandler = () => {
    history.back();
  };

  const saveUpdateButtonHandler = (values: {
    uId: string | undefined;
    profile: {
      email: string;
      phoneNumber: string;
      password: string;
      firstName: string;
      lastName: string;
      statusId: string;
      photoURL: string;
    };
    address:
      | {
          addressLineOne: string;
          streetName: string;
          houseNumber: string;
          postalCode: string;
          cityName: string;
        }[]
      | undefined;
    carDetails:
      | {
          carId: string;
        }[]
      | undefined;
  }) => {
    if (isEditable) {
      const params = JSON.stringify(values);
      const formatJson = params.replace(/'/g, '\'\'');
      dispatch(updateCustomerRequest(formatJson));
    } else {
      const params = JSON.stringify(_.omit(values, 'uId'));
      const formatJson = params.replace(/'/g, '\'\'');
      dispatch(createCustomerRequest(formatJson));
    }
  };

  useEffect(() => {
    if (statusList.length === 0) {
      dispatch(getStatusDetails());
    } else {
      setStatusDropdownData(statusList);
    }
  }, [statusList]);

  const handleCheckedCars = (
    e: React.ChangeEvent<HTMLInputElement>,
    carInfo: CarIProps
  ) => {
    if (isEditable) {
      if (e.target.checked) {
        tempEditCheckedCars.push(carInfo);
      } else {
        _.remove(
          tempEditCheckedCars,
          (obj: CarIProps) => obj.carId === carInfo.carId
        );
      }
    } else {
      if (e.target.checked) {
        tempCheckedCars.push(carInfo);
        const carToUpdate = _.find(unlinkedCars, { 'Number Plate': carInfo['Number Plate'] });
        if (carToUpdate) {
          carToUpdate.checked = true;
          setUnLinkedCars([...unlinkedCars]);
        }
      } else {
        const carToUpdate = _.find(unlinkedCars, { 'Number Plate': carInfo['Number Plate'] });
        if (carToUpdate) {
          carToUpdate.checked = false;
          setUnLinkedCars([...unlinkedCars]);
        }
        _.remove(
          tempCheckedCars,
          (obj: CarIProps) => obj.carId === carInfo.carId
        );
      }
    }
  };

  const handleEditPageCheckedCars = (
    e: React.ChangeEvent<HTMLInputElement>,
    carInfo: CarIProps
  ) => {
    if (e.target.checked) {
      tempCheckedCars.push(carInfo);
    } else {
      const carToUpdate = _.find(unlinkedCars, { 'Number Plate': carInfo['Number Plate'] });
        if (carToUpdate) {
          carToUpdate.checked = false;
          setUnLinkedCars([...unlinkedCars]);
        }
      _.remove(
        tempCheckedCars,
        (obj: CarIProps) => obj.carId === carInfo.carId
      );
      setTempCheckedCars(tempCheckedCars);
    }
  };

  const handleSave = () => {
    if (isEditable) {
      setEditCheckedCars(tempEditCheckedCars);
    } else {
      const filetredCars =  _.filter(unlinkedCars, (obj) => obj.checked === true);
      setCheckedCars(filetredCars);
    }
  };

  const handleReset = () => {
    setPhotoURL(isEditable ? viewCustomerResponseData.details[0].profile.photoURL : '');
    //setCheckedCars([]);
    //setTempCheckedCars([]);
    setEditCheckedCars([]);
    const setChecked = _.map(
      viewCustomerResponseData.details[0].cars,
      (obj) => {
        return { ...obj, checked: true };
      }
    );
    setEditCheckedCars(setChecked);
    setTempEditCheckedCars(setChecked);
    setTempCheckedCars(setChecked);
    setCheckedCars(setChecked);
  };
  
  if (access !== '') {
    if (
      access === AccessTypes.fullAccess ||
      access === AccessTypes.readOnlyAccess
    ) {
      return (
        <ManageCustomers
          backButtonHandler={backButtonHandler}
          customerData={customerData}
          statusDropDownData={statusDropdownData}
          isDisable={isDisable}
          isEditable={isEditable}
          handleUserDetailsCollapse={handleUserDetailsCollapse}
          handleCarsDetailsCollapse={handleCarsDetailsCollapse}
          userDetailsCollapse={userDetailsCollapse}
          carsDetailsCollapse={carsDetailsCollapse}
          saveUpdateButtonHandler={saveUpdateButtonHandler}
          passwordType={passwordType}
          handlePasswordTypeChange={handlePasswordTypeChange}
          isCreate={isCreate}
          handleModalClose={handleModalClose}
          handleModalOpen={handleModalOpen}
          modalShow={modalShow}
          unlinkedCars={unlinkedCars}
          linkedCars={linkedCars}
          checkedCars={checkedCars}
          handleCheckedCars={handleCheckedCars}
          handleEditPageCheckedCars={handleEditPageCheckedCars}
          handleSave={handleSave}
          handleReset={handleReset}
          editCheckedCars={editCheckedCars}
          onDrop={onDrop}
          photoURL={photoURL}
          searchValue={searchValue}
          handleSearch={handleSearch}
        />
      );
    } else {
      return <NotFound />;
    }
  } else {
    return <Loader />;
  }
};
