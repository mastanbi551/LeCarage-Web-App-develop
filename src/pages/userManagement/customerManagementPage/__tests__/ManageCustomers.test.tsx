import React from 'react';
import { render } from '@testing-library/react';
import { ManageCustomers } from '../manageCustomers/ManageCustomers';

describe('ManageCustomers component', () => {
  const mockProps = {
    backButtonHandler: jest.fn(),
    statusDropDownData: [{ label: 'Active', value: 'active' }],
    isDisable: false,
    isEditable: true,
    handleUserDetailsCollapse: jest.fn(),
    handleCarsDetailsCollapse: jest.fn(),
    handlePaymentDetailsCollapse: jest.fn(),
    userDetailsCollapse: false,
    carsDetailsCollapse: false,
    paymentDetailsCollapse: false,
    saveUpdateButtonHandler: jest.fn(),
    passwordType: 'password',
    handlePasswordTypeChange: jest.fn(),
    customerData: {
      uId: '1',
      profile: {
        email: 'test@example.com',
        phoneNumber: '1234567890',
        password: 'test123',
        firstName: 'Test',
        lastName: 'User',
        statusId: 'active',
      },
      address: [
        {
          addressLineOne: '123 Main St',
          streetName: 'Main St',
          houseNumber: '123',
          postalCode: '12345',
          cityName: 'Anytown',
        },
      ],
      carDetails: [
        {
          carId: '1',
        },
      ],
    },
    isCreate: false,
    handleModalClose: jest.fn(),
    handleModalOpen: jest.fn(),
    modalShow: false,
    unlinkedCars: [],
    linkedCars: [],
    checkedCars: [],
    handleCheckedCars: jest.fn(),
    handleEditPageCheckedCars: jest.fn(),
    handleSave: jest.fn(),
    handleReset: jest.fn(),
    tempEditCheckedCars: [],
    editCheckedCars: [],
  };

  it('should render correctly', () => {
    const { container } = render(<ManageCustomers {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
