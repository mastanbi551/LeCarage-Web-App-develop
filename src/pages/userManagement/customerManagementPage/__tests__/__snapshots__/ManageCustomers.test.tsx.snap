// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageCustomers component should render correctly 1`] = `
<div>
  <div
    class="page-container-details"
  >
    <div
      class="page-container-details__header-content"
    >
      <svg
        class="back-icon"
        data-testid="back-icon"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="19"
          x2="5"
          y1="12"
          y2="12"
        />
        <polyline
          points="12 19 5 12 12 5"
        />
      </svg>
      <h1
        class="page-container-details__header-content__heading"
      >
        Customer
      </h1>
    </div>
    <form
      class="container-fluid"
    >
      <div
        class="row "
      >
        <div
          class="col-md-8 manage-customers-page"
        >
          <div
            class="col-md-12"
          >
            <div
              class="details-container"
            >
              <div
                class="details-container__details-container-header"
              >
                <h1
                  class="details-container__details-container-header__heading"
                >
                  Customer Details
                </h1>
                <svg
                  class="details-container__details-container-header__collapse-icon"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                  />
                </svg>
              </div>
              <div
                class="collapse"
              >
                <div
                  class="manage-customers-page__customer-info"
                >
                  <div
                    class="row pdtopbtm"
                  >
                    <div
                      class="col-md-6"
                    >
                      <label
                        class="label"
                      >
                        First Name
                      </label>
                      <div
                        class="input-text-field"
                      >
                        <input
                          autocomplete="new-password"
                          autocorrect="off"
                          class="firstName static"
                          data-testid="input-field"
                          id="firstname_field"
                          placeholder=""
                          required=""
                          type="text"
                          value="Test"
                        />
                      </div>
                    </div>
                    <div
                      class="col-md-6"
                    >
                      <label
                        class="label"
                      >
                        Last Name
                      </label>
                      <div
                        class="input-text-field"
                      >
                        <input
                          autocomplete="new-password"
                          autocorrect="off"
                          class="lastName static"
                          id="lastname_field"
                          placeholder=""
                          required=""
                          type="text"
                          value="User"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="row pdtopbtm"
                  >
                    <div
                      class="col-md-6"
                    >
                      <label
                        class="label"
                      >
                        Email Address
                      </label>
                      <div
                        class="input-text-field"
                      >
                        <input
                          autocomplete="new-password"
                          autocorrect="off"
                          class="email static"
                          id="email_field"
                          placeholder=""
                          required=""
                          type="text"
                          value="test@example.com"
                        />
                      </div>
                    </div>
                    <div
                      class="col-md-6"
                    >
                      <label
                        class="label"
                      >
                        CellPhone Number
                      </label>
                      <div
                        class="input-text-field"
                      >
                        <input
                          autocomplete="new-password"
                          autocorrect="off"
                          class="phoneNumber static"
                          id="cell-phone_field"
                          placeholder=""
                          required=""
                          type="text"
                          value="1234567890"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="row pdtopbtm"
                  >
                    <div
                      class="col-md-6 customerspassword"
                    >
                      <label
                        class="label"
                      >
                        Password
                      </label>
                      <div
                        class="input-text-field"
                      >
                        <input
                          autocomplete="new-password"
                          autocorrect="off"
                          class="password password-field static"
                          disabled=""
                          id="password_field"
                          placeholder=""
                          required=""
                          type="password"
                          value="test123"
                        />
                        <span
                          class="multiple-icons"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="details-card-containers__user-address-container"
                  >
                    <h1
                      class="details-card-containers__user-address-container__heading"
                    >
                      Customer Address
                    </h1>
                    <div
                      class="details-card-containers__user-address-container__address-details"
                    >
                      <div>
                        <div
                          class="details-card-containers__user-address-container__address-details__delete-icon"
                        >
                          <svg
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g>
                              <path
                                d="M0 0h24v24H0z"
                                fill="none"
                              />
                              <path
                                d="M17 6h5v2h-2v13a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V8H2V6h5V3a1 1 0 0 1 1-1h8a1 1 0 0 1 1 1v3zm1 2H6v12h12V8zm-9 3h2v6H9v-6zm4 0h2v6h-2v-6zM9 4v2h6V4H9z"
                              />
                            </g>
                          </svg>
                        </div>
                        <div
                          class="row pdtopbtm"
                        >
                          <div
                            class="col-md-6"
                          >
                            <label
                              class="label"
                            >
                              Address Type
                            </label>
                            <div
                              class="input-text-field"
                            >
                              <input
                                autocomplete="new-password"
                                autocorrect="off"
                                class=" static"
                                id="address_field"
                                placeholder=""
                                required=""
                                type="text"
                                value="123 Main St"
                              />
                            </div>
                          </div>
                          <div
                            class="col-md-6"
                          >
                            <label
                              class="label"
                            >
                              Street Name
                            </label>
                            <div
                              class="input-text-field"
                            >
                              <input
                                autocomplete="new-password"
                                autocorrect="off"
                                class=" static"
                                id="street_field"
                                placeholder=""
                                required=""
                                type="text"
                                value="Main St"
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class="row pdtopbtm"
                        >
                          <div
                            class="col-md-6"
                          >
                            <label
                              class="label"
                            >
                              House Apartment no
                            </label>
                            <div
                              class="input-text-field"
                            >
                              <input
                                autocomplete="new-password"
                                autocorrect="off"
                                class=" static"
                                id="houseapratment_field"
                                placeholder=""
                                required=""
                                type="text"
                                value="123"
                              />
                            </div>
                          </div>
                          <div
                            class="col-md-6"
                          >
                            <label
                              class="label"
                            >
                              Postal Code
                            </label>
                            <div
                              class="input-text-field"
                            >
                              <input
                                autocomplete="new-password"
                                autocorrect="off"
                                class=" static"
                                id="postalcode_field"
                                placeholder=""
                                required=""
                                type="text"
                                value="12345"
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class="row pdtopbtm"
                        >
                          <div
                            class="col-md-6"
                          >
                            <label
                              class="label"
                            >
                              City
                            </label>
                            <div
                              class="input-text-field"
                            >
                              <input
                                autocomplete="new-password"
                                autocorrect="off"
                                class=" static"
                                id="streetname_field"
                                placeholder=""
                                required=""
                                type="text"
                                value="Anytown"
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class="details-card-containers__user-address-container__address-details__add-Address"
                        >
                          <button
                            class="add-button"
                            type="button"
                          >
                            ADD ADDRESS
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="details-container"
            >
              <div
                class="details-container__details-container-header"
              >
                <h1
                  class="details-container__details-container-header__heading"
                >
                  Cars Linked
                </h1>
                <svg
                  class="details-container__details-container-header__collapse-icon"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                  />
                </svg>
              </div>
              <div
                class="collapse"
              >
                <div
                  class="details-card-containers__cars-container__cars-details"
                >
                  <div
                    class="modal-view"
                  />
                  <div
                    class="container"
                  >
                    <table
                      class="table table-bordered table-striped table-hover"
                    >
                      <thead>
                        <tr>
                          <th>
                            Number Plate
                          </th>
                          <th>
                            Make
                          </th>
                          <th>
                            Model
                          </th>
                          <th>
                            Color
                          </th>
                        </tr>
                      </thead>
                      <tbody />
                    </table>
                  </div>
                  <div
                    class="details-card-containers__cars-container__cars-details__add-car"
                  >
                    <button
                      class="add-button"
                      type="button"
                    >
                      Link CAR
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-md-4 details-card-containers"
        >
          <div
            class="col-md-12"
          >
            <div
              class="card-container"
            >
              <div
                class="col-md-12"
              >
                <div
                  class="card-container__image-wrapper"
                >
                  <div
                    class="card-container__image-container"
                  >
                    <img
                      class="card-container__image-container__image"
                      style="object-fit: cover;"
                    />
                    <h1
                      class="card-container__image-container__heading"
                    />
                  </div>
                </div>
                <div
                  class="card-container__file-upload-container"
                >
                  <h1
                    class="card-container__heading"
                  >
                    Upload Image
                  </h1>
                  <div
                    class="outer"
                  >
                    <div
                      class="imgload"
                    >
                       
                    </div>
                    <div
                      class="DropZone"
                      role="button"
                      tabindex="0"
                    >
                      Click or drop your file here
                    </div>
                  </div>
                </div>
                 
              </div>
              <h1
                class="card-container__heading"
              >
                Customer Actions
              </h1>
              <div
                class="dropdown input-text-field"
              >
                <label
                  class="dropDown-label"
                >
                  Set Status
                </label>
                <select
                  class="custom-select status-dropdown"
                >
                  <option
                    class="custom-option"
                    value="active"
                  >
                    <span
                      class="inactive-icon"
                    />
                    <span>
                      Active
                    </span>
                  </option>
                </select>
              </div>
              <div
                class="line"
              />
              <button
                class="greenButton w-100"
                type="button"
              >
                Save & Update
              </button>
              <button
                class="add-button w-100"
                type="button"
              >
                Discard
              </button>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
